pkg/dockercreds/k8sdockercreds/k8s_keychain.go

```diff
@@ -41,35 +41,24 @@ func (f *k8sSecretKeychainFactory) KeychainForSecretRef(ctx context.Context, ref
 		return authn.NewMultiKeychain(f.volumeKeychain, keychain), nil // k8s keychain with no secrets
 	}
 
-	serviceAccountKeychain, err := keychainFromServiceAccount(ctx, ref, &secret.Fetcher{Client: f.client})
+	serviceAccountKeychain, err := f.KeychainFromServiceAccountSecrets(ctx, ref)
 	if err != nil {
 		return nil, err
 	}
 
-	k8sKeychain, err := k8schain.New(ctx, f.client, k8schain.Options{
-		Namespace:          ref.Namespace,
-		ServiceAccountName: ref.ServiceAccount,
-		ImagePullSecrets:   toStringPullSecrets(ref.ImagePullSecrets),
-	})
+	k8sKeychain, err := f.keychainFromServiceAccountPullSecrets(ctx, ref)
 	if err != nil {
 		return nil, err
 	}
 
 	return authn.NewMultiKeychain(serviceAccountKeychain, f.volumeKeychain, k8sKeychain), nil
 }
 
-func toStringPullSecrets(secrets []corev1.LocalObjectReference) []string {
-	var stringSecrets []string
-	for _, s := range secrets {
-		stringSecrets = append(stringSecrets, s.Name)
-	}
-	return stringSecrets
-}
-
-func keychainFromServiceAccount(ctx context.Context, secretRef registry.SecretRef, fetcher *secret.Fetcher) (authn.Keychain, error) {
+func (f *k8sSecretKeychainFactory) KeychainFromServiceAccountSecrets(ctx context.Context, ref registry.ServiceAccountRef) (authn.Keychain, error) {
```

Comment:
It looks like we are exposing a public api here to avoid the k8schain which I fear might be exposing too much implementation detail. Could we instead provide a keychain that treats the k8schain as the lowest priority?